include/dense_map.h

```diff
@@ -17,7 +18,9 @@ class dense_map : private std::vector<Value> {
 
 	explicit dense_map(const size_t size) : vector(size) {}
 
-	template <typename InputIterator>
+	explicit dense_map(const size_t size, const Value& init) : vector(size, init) {}
+
+	template <typename InputIterator, typename std::iterator_traits<InputIterator>::pointer Enable = nullptr>
```

Comment:
That `InputIterator` is an input iterator - otherwise this overload would be ambiguous with the one I added above. Thankfully there's now the `std::input_iterator` concept.